src/NetTopologySuite/IO/WKTReader.cs

```diff
@@ -104,17 +98,16 @@ public bool IsOldNtsMultiPointSyntaxAllowed
         /// </summary>
         public GeometryFactory Factory
         {
-            get => _ntsGeometryServices.CreateGeometryFactory(_precisionModel, DefaultSRID, _coordinateSequencefactory);
+            get => _ntsGeometryServices.CreateGeometryFactory(DefaultSRID);
             set
             {
                 if (value != null)
                 {
-                    if (value.GeometryOverlay != _ntsGeometryServices.GeometryOverlay)
-                        throw new ArgumentException("Factory has incompatible set of overlay functions assigned.", nameof(value));
-
-                    _coordinateSequencefactory = value.CoordinateSequenceFactory;
-                    _precisionModel = value.PrecisionModel;
-                    DefaultSRID = value.SRID;
+                    _ntsGeometryServices = (NtsGeometryServices)Activator.CreateInstance(_ntsGeometryServices.GetType(),
+                        value.CoordinateSequenceFactory, value.PrecisionModel,
+                        value.SRID, value.GeometryOverlay);
+                    // Not sure about this:
+                    // DefaultSRID = value.SRID;
```

Comment:
why is this different from what we do in the `GeometryFactory` constructor?

Reply:
Because otherwise `GeometryFactoryExTest` would fail.
It is not possible to configure `NtsGeometryServices` to return anything other than `GeometryFactory`.

We could catch the Exception and provide necessary information.